### PR TITLE
SAK-46600 Conversations: Fixed aria for comment menu

### PR DESCRIPTION
**Pull request by adrianfish**
https://sakaiproject.atlassian.net/browse/SAK-46600

### CODE REVIEW

File: webcomponents/tool/src/main/frontend/js/conversations/sakai-comment.js [hunk 1]
```diff
@@ -67,9 +67,15 @@ export class SakaiComment extends SakaiElement {
...
-        <a slot="trigger" id="options-menu-link-${this.comment.id}" href="javascript:;">
-          <div><sakai-icon type="menu" size="small"></sakai-icon></div>
-        </a>
+        <div slot="trigger">
+          <a href="javascript:;"
+              title="${this.i18n.comment_options_menu_tooltip}"
+              aria-haspopup="true"
+              aria-label="${this.i18n.comment_options_menu_tooltip}">
+            <sakai-icon type="menu" size="small"></sakai-icon>
+          </a>
```
Comment (ottenhoff) [inline webcomponents/tool/src/main/frontend/js/conversations/sakai-comment.js:76]: this seems fine but when i see href="javascript" i get kind of suspicious.

why not a <button>?

Comment (adrianfish) [reply webcomponents/tool/src/main/frontend/js/conversations/sakai-comment.js:76]: I use "javascript:;" all the time as I don't want the page hopping to the top through using a #. What's the better solution?

Comment (ottenhoff) [reply webcomponents/tool/src/main/frontend/js/conversations/sakai-comment.js:76]: button

Comment (adrianfish) [reply webcomponents/tool/src/main/frontend/js/conversations/sakai-comment.js:76]: Alright, I'll switch this to button. What about when things have to be links though? Transparent button?

Comment (adrianfish) [reply webcomponents/tool/src/main/frontend/js/conversations/sakai-comment.js:76]: I agree entirely. A "nude" button would do the trick.

Comment (adrianfish) [reply webcomponents/tool/src/main/frontend/js/conversations/sakai-comment.js:76]: I don't know if I like that commit I just pushed. It's more !importants and I know we don't want those.

**Review by ottenhoff — APPROVED** [review]
My only real comment: it would be awesome to have this style-less button available everywhere in Sakai

**Comment by SedueRey** [comment]
C'mon ... more importants?????

If you want this kind of behaviour, Bootstrap 3.3.7 provides you btn btn-link and it will work all over Sakai

@ottenhoff, @adrianfish this shouldn't be merged with `!important`, please no. There are ways to do it always better than  important.

Even if you want a new class, you can use `unset`. Whatever before !important and without reviews.

**Comment by adrianfish** [comment]
It was reviewed. Sam didn't like the anchor I started with, understandably, so I swapped it for a button. I didn't want to mess with our button styles so I used important. Just pragmatism. Please open another Jira to add a new transparent button class - I was going to do that, but you may as well. You seem to know more about it than me anyway. I am aware of the problems with important, just so you know.

**Comment by SedueRey** [comment]
Well ... Pragmatism is not a good answer for good software but whatever, we will keep fixing bugs here and there, sure.

**Comment by adrianfish** [comment]
Just open another jira about a transparent button, put some comments on and assign to Shawn.

Leaving aside your incredibly bad manners @SedueRey, I'll try and explain my ethos on this ticket:

1. This was effectively a blocker for screen reader users.
2. Our button css is complex and I didn't want to produce any more bugs by messing with it. I figured the use of a button (semantic), the fixing of a bad a11y bug and the avoidance of disturbing the existing button styles justified the use of an !important in this case. I did intend to open a jira for Shawn or Mike to look at, but I'm not as disciplined as you so I forgot. Please accept my apologies in that regard.

You keep fixing those bugs :)

**Comment by SedueRey** [comment]
Well, I'm going to leave it here but typing it in an online translator to be sure I don't think I'm saying what I'm not saying. I encourage you to try to write in a language that is not yours and make yourself understood to be considered rude.

The facts are:

1. While we were fixing the !important and fixed colors in another of the PRs, which were affecting all of Sakai by not having the namespace, you have added this PR, it has been validated and merged. The rest of us usually have this debate in core calls.

2. That there is a blocker is not an excuse to put bad code, !important is bad code no matter when you look at it. Screen reader or mobile or whatever another platform.

3. With all due respect, @adrianfish , you are not a newbie to Sakai. You have been around as long or longer than me, you are a member of the PMC, which I am not and you should know the libraries we have, in what state and that we try to improve the code every time something new is added. I repeat: you do NOT need to add a new class. You should know that.

After this, I stand by it, pragmatism or haste is no excuse for bad code. If you want, we debate it in the core call, I have no problem about it.

This PR should not have been mixed and validated by @ottenhoff . Period.

Thanks. That said, I encourage you to discuss it on the next core call.

**Comment by adrianfish** [comment]
I understand that you're writing in a second language and I respect this deeply, but I also recognise sarcasm when I see it.

"we will keep fixing bugs here and there, sure." is just sarcastic and there's no need. It was reviewed, I made amendments and made a value judgment.

If you open a JIRA ticket and detail your concerns and a way to address them, I'll put the fix in.

**Comment by SedueRey** [comment]
No, @adrianfish, that's was not sarcastic. When I wrote that I wanted to mean, really,  that we will keep fixing all bugs that we find, we're not going to step back. We are part of this community and want to install 22 this july so, literally, we will keep fixing bugs where we would find them, if you prefer it in that way.

That was not a sarcasm.

**Comment by adrianfish** [comment]
In that case I apologise Eduardo. That reads really sarcastically in english but perhaps I should have been more understanding. Maybe you can tell me in spanish next time (seriously). I can read spanish reasonably well. That may have been clearer to me :)

Anyway, can you open a JIRA and assign to me? I'll see what I can do with it. Any any suggestions on that JIRA will be welcome.

**Pull request merged**